### DISCUSSION TITLE
Birt Jdbc Connectivity error

### DISCUSSION THREAD

**Discussion by manikandanrajavel** [eclipse-birt/birt#1984] [Q&A] [answered]
I am getting below error in when try to run the BIRT report in browser. some reason BIRT taking system user id as DB login user. how do i configure jdbc for BIRT. It is java based web application, deployed in tomcat. which file jdbc needs to be configured for BIRT? why is it picking system user id as DB user id. 
 
Any help would be greatly appreciated.

?xml version="1.0" encoding="UTF-8"?><soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"><soapenv:Body><soapenv:Fault><faultcode>BirtRunReportActionHandler.__execute()</faultcode><faultstring>org.eclipse.birt.report.engine.api.EngineException: An exception occurred during processing. Please see the following message for details:
Cannot open the connection for the driver: org.eclipse.birt.report.data.oda.jdbc.
    org.eclipse.birt.report.data.oda.jdbc.JDBCException: There is an error in get connection, [jcc][4038][12241][4.27.25] T2LUW exception: SQL1092N  The requested command or operation failed because the user ID does not have the authority to perform the requested command or operation.  User ID: &quot;XXXXXX&quot;.
 ERRORCODE=-1092, SQLSTATE=     .</faultstring><detail><ns1:hostname xmlns:ns1="http://xml.apache.org/axis/">XXXXXXXXX</ns1:hostname><string>org.eclipse.birt.report.engine.api.EngineException: An exception occurred during processing. Please see the following message for details:
Cannot open the connection for the driver: org.eclipse.birt.report.data.oda.jdbc.
    org.eclipse.birt.report.data.oda.jdbc.JDBCException: There is an error in get connection, [jcc][4038][12241][4.27.25] T2LUW exception: SQL1092N  The requested command or operation failed because the user ID does not have the authority to perform the requested command or operation.  User ID: &quot;XXXXXXX&quot;.
 ERRORCODE=-1092, SQLSTATE=     .
	at org.eclipse.birt.report.engine.executor.ExecutionContext.addException(ExecutionContext.java:1060)
	at org.eclipse.birt.report.engine.executor.ExecutionContext.addException(ExecutionContext.java:1045)
	at org.eclipse.birt.report.engine.executor.QueryItemExecutor.executeQuery(QueryItemExecutor.java:83)
	at org.eclipse.birt.report.engine.executor.TableItemExecutor.execute(TableItemExecutor.java:60)
	at org.eclipse.birt.report.engine.internal.executor.wrap.WrappedReportItemExecutor.execute(WrappedReportItemExecutor.java:45)
	at org.eclipse.birt.report.engine.internal.executor.emitter.ReportItemEmitterExecutor.execute(ReportItemEmitterExecutor.java:45)

**Comment by speckyspooky** [comment]
Could you provide a screen of your added data source connection for the used JDBC-driver.
Good to know is which data base / data source should be connected Oracle, MS SQL Server, MySQL, etc.
and which driver you have used here.

It sounds like DB2 based on your error code according to the link: [https://www.ibm.com/support/pages/getting-sql1092n-error-when-running-any-db2-commands-domain-id](url)

**Reply by manikandanrajavel** [reply, marked as answer]
Thank you for your reply !, I am using DB2 . Resolved the issue now. Fix was tomcat/conf/context.xml defined the DB2 datasource instead of server.xml.
